File: testing.py
from flask import Flask, request, jsonify
import sqlite3

app = Flask(__name__)

# Database initialization
def init_db():
    conn = sqlite3.connect('items.db')
    cursor = conn.cursor()
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS items (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT
        )
    ''')
    conn.commit()
    conn.close()

init_db()

def get_db_connection():
    conn = sqlite3.connect('items.db')
    conn.row_factory = sqlite3.Row  # Return rows as dictionaries
    return conn

@app.route('/items', methods=['GET'])
def get_items():
    conn = get_db_connection()
    items = conn.execute('SELECT * FROM items').fetchall()
    conn.close()
    return jsonify([dict(item) for item in items])

@app.route('/items/<int:item_id>', methods=['GET'])
def get_item(item_id):
    conn = get_db_connection()
    item = conn.execute('SELECT * FROM items WHERE id = ?', (item_id,)).fetchone()
    conn.close()
    if item is None:
        return jsonify({'error': 'Item not found'}), 404
    return jsonify(dict(item))

@app.route('/items', methods=['POST'])
def create_item():
    data = request.get_json()
    if not data or 'name' not in data:
        return jsonify({'error': 'Name is required'}), 400

    name = data['name']
    description = data.get('description', None)

    conn = get_db_connection()
    cursor = conn.cursor()
    cursor.execute('INSERT INTO items (name, description) VALUES (?, ?)', (name, description))
    conn.commit()
    item_id = cursor.lastrowid
    conn.close()

    return jsonify({'id': item_id, 'name': name, 'description': description}), 201

@app.route('/items/<int:item_id>', methods=['PUT'])
def update_item(item_id):
    data = request.get_json()
    if not data or 'name' not in data:
        return jsonify({'error': 'Name is required'}), 400

    name = data['name']
    description = data.get('description', None)

    conn = get_db_connection()
    cursor = conn.cursor()
    cursor.execute('UPDATE items SET name = ?, description = ? WHERE id = ?', (name, description, item_id))
    conn.commit()
    conn.close()

    return jsonify({'id': item_id, 'name': name, 'description': description})

@app.route('/items/<int:item_id>', methods=['DELETE'])
def delete_item(item_id):
    conn = get_db_connection()
    cursor = conn.cursor()
    cursor.execute('DELETE FROM items WHERE id = ?', (item_id,))
    conn.commit()
    conn.close()
    return '', 204

if __name__ == '__main__':
    app.run(debug=True)

File: testing.php
<?php

$db_host = 'localhost';
$db_name = 'mydatabase';
$db_user = 'postgres';
$db_pass = 'your_postgres_password';
$db_port = '5432';

try {
    // Connect to PostgreSQL using PDO
    $pdo = new PDO("pgsql:host=$db_host;port=$db_port;dbname=$db_name", $db_user, $db_pass);
    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    // Create table if it doesn't exist
    $pdo->exec("CREATE TABLE IF NOT EXISTS items (
        id SERIAL PRIMARY KEY,
        name VARCHAR(255) NOT NULL,
        description TEXT
    )");

    function createItem($pdo, $name, $description) {
        $stmt = $pdo->prepare("INSERT INTO items (name, description) VALUES (:name, :description)");
        $stmt->bindParam(':name', $name);
        $stmt->bindParam(':description', $description);
        $stmt->execute();
        return $pdo->lastInsertId();
    }

    // Read (GET) - single item
    function getItem($pdo, $id) {
        $stmt = $pdo->prepare("SELECT * FROM items WHERE id = :id");
        $stmt->bindParam(':id', $id, PDO::PARAM_INT);
        $stmt->execute();
        return $stmt->fetch(PDO::FETCH_ASSOC);
    }

    // Read (GET) - all items
    function getItems($pdo) {
        $stmt = $pdo->query("SELECT * FROM items");
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    // Update (PUT)
    function updateItem($pdo, $id, $name, $description) {
        $stmt = $pdo->prepare("UPDATE items SET name = :name, description = :description WHERE id = :id");
        $stmt->bindParam(':id', $id, PDO::PARAM_INT);
        $stmt->bindParam(':name', $name);
        $stmt->bindParam(':description', $description);
        $stmt->execute();
        return $stmt->rowCount(); // Returns number of affected rows
    }

    // Delete (DELETE)
    function deleteItem($pdo, $id) {
        $stmt = $pdo->prepare("DELETE FROM items WHERE id = :id");
        $stmt->bindParam(':id', $id, PDO::PARAM_INT);
        $stmt->execute();
        return $stmt->rowCount();
    }

    // Example Usage (Simulated HTTP request handling)
    if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_POST['name'])) {
        $id = createItem($pdo, $_POST['name'], $_POST['description'] ?? null);
        echo json_encode(['id' => $id, 'message' => 'Item created']);
    } elseif ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['id'])) {
        $item = getItem($pdo, $_GET['id']);
        echo json_encode($item);
    } elseif ($_SERVER['REQUEST_METHOD'] === 'GET') {
        $items = getItems($pdo);
        echo json_encode($items);
    } elseif ($_SERVER['REQUEST_METHOD'] === 'PUT' && isset($_GET['id'])) {
        parse_str(file_get_contents("php://input"), $_PUT_VARS);
        $updated = updateItem($pdo, $_GET['id'], $_PUT_VARS['name'], $_PUT_VARS['description'] ?? null);
        echo json_encode(['updated' => $updated, 'message' => 'Item updated']);
    } elseif ($_SERVER['REQUEST_METHOD'] === 'DELETE' && isset($_GET['id'])) {
        $deleted = deleteItem($pdo, $_GET['id']);
        echo json_encode(['deleted' => $deleted, 'message' => 'Item deleted']);
    }

} catch (PDOException $e) {
    echo "Database Error: " . $e->getMessage();
}

?>